uwblueprint/community-fridge-kw
# feat: fix recurring end date upon deletion

## Brief description. What is this change? 

### [Cancel All Following Donations, End Date Still the Same](https://www.notion.so/uwblueprintexecs/39bf3fa8ccce4beb93205438181ec01a?v=46a37f121c4249a5baf2041e84425e82&p=892fd1e9a13f45d89602b9186c88a81b)



## Implementation description. How did you make this change?
* Found maximum date remaining after deletion and updated all objects with the same ```recurring_donation_id``` to have this updated date -> found this to be the simplest solution because the frequency makes the ability to just subtract the date to be more difficult /  more code. Sequelize has a nice function for this.
* Some error prevention in the sense that you cannot update the end recurring date if the object with that ```recurring_donation_id``` no longer exists. I used the count function for this, but am open to quicker / more efficient methods here.
* Refactored common test objects to be under a ```testUtils``` folder but not sure if I placed correctly (we discussed this in the first recurring deletion backend tickets.
* Added one more test for this use case and as a result had to add more objects to the ```testSchedules``` object



## Steps to test
1. Create a recurring donation 
2. Try to delete "this and following recurring donation" on of the donations in the middle in time of the beginning to end date. Remember this date. Refresh donation page and make sure it is one day less than the date you remembered.
3. Can delete the very first item and make sure that no error is thrown (for point 2 under implementation details)




## Checklist
